### PR TITLE
resolve dev to fitchRoute.

### PR DESCRIPTION
**Pull request by krasyukden**

### CODE REVIEW

File: src/ComicsPage.module.css [hunk 1]
```diff
@@ -0,0 +1,48 @@
+*{
+	margin: 0;
```
Comment (alinalaktionova) [inline src/ComicsPage.module.css]: tabs

File: src/ComicsPage.tsx [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  match: {
+    params: {
+      charachterId: string;
+    }
+  },
+  loading: boolean
+}
+
+interface ComicsState {
+  comics: any[],
```
Comment (alinalaktionova) [inline src/ComicsPage.tsx]: remove any

File: src/ComicsPage.tsx [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+  constructor(props: ComicsProps) {
+    super(props)
+    this.state = {
+      comics: [],
+      loading: true
+      }
+  }
+
+  componentDidMount(): void {
+        const characterId = this.props?.match?.params?.charachterId;
```
Comment (alinalaktionova) [inline src/ComicsPage.tsx]: again tab

File: src/ComicsPage.tsx [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    getComicsByCharacter(characterId).then((comics: any) => {
+      this.setState({ 
+        comics,
+        loading: false 
+      });
+    })
+  }
+
+  render() {
+     return <>{ }
```
Comment (alinalaktionova) [inline src/ComicsPage.tsx]: ????

File: src/ComicsPage.tsx [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+      {this.state.loading ? <Preloader /> :
+      <div className={styles.wrapper}>
```
Comment (alinalaktionova) [inline src/ComicsPage.tsx]: use destructuring

File: src/ComicsPage.tsx [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+        <div className={styles.heroTitle}>Hero comics</div>
+        <div className={styles.wrapperHeroes}>
+          {this.state.comics.slice(0, 5).map(comics => {
```
Comment (alinalaktionova) [inline src/ComicsPage.tsx]: why do you need slice?

File: src/ComicsPage.tsx [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+            return <div className={styles.hero} key={comics.id}>
+              <img className={styles.heroesImg} src={comics.thumbnail.path != null
```
Comment (alinalaktionova) [inline src/ComicsPage.tsx]: use strict equality

File: src/ComicsPage.tsx [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+                ? `${comics.thumbnail.path}.${comics.thumbnail.extension}` : 'Photo'} />
```
Comment (alinalaktionova) [inline src/ComicsPage.tsx]: again destructuring is better here

File: src/HomePage.module.css [hunk 9]
```diff
@@ -0,0 +1,72 @@
+*{
+	margin: 0;
```
Comment (alinalaktionova) [inline src/HomePage.module.css]: code style

File: src/ComicsPage.module.css [hunk 10]
```diff
@@ -0,0 +1,48 @@
+*{
```
Comment (alinalaktionova) [inline src/ComicsPage.module.css]: better to name file comics_page.style.css

File: src/HomePage.tsx [hunk 11]
```diff
@@ -0,0 +1,143 @@
...
+interface ComicsProps {
+  match: {
+    params: {
+      charachterId: string;
+    }
+  }
+}
+
+interface HomeProps {
+  heroes: any[],
```
Comment (alinalaktionova) [inline src/HomePage.tsx]: any

File: src/HomePage.tsx [hunk 12]
```diff
@@ -0,0 +1,143 @@
...
+
+interface ComicsProps {
+  match: {
+    params: {
+      charachterId: string;
+    }
+  }
+}
+
+interface HomeProps {
```
Comment (alinalaktionova) [inline src/HomePage.tsx:20]: why HomeState and HomeProps has the same properties? And what means 'Home'?

File: src/HomePage.tsx [hunk 13]
```diff
@@ -0,0 +1,143 @@
...
+          inputValue: nameCharacter,
+          heroes: heroes,
+          loading: false
+
+        })
+      })
+    } else {
+
+      getCharacters().then((heroes: any) => {
+
```
Comment (alinalaktionova) [inline src/HomePage.tsx:69]: empty line for what?

File: src/HomePage.tsx [hunk 14]
```diff
@@ -0,0 +1,143 @@
...
+      loading: true,
+      inputValue: ''
+
+    }
+
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+
+  }
+  loadFromServer() {
```
Comment (alinalaktionova) [inline src/HomePage.tsx:51]: method name should be clear and describes what you really do

File: src/HomePage.tsx [hunk 15]
```diff
@@ -0,0 +1,143 @@
...
+          <div className={styles.wrapperHeroes}>
+            {this.state.heroes.slice(0, 5).map((heroes: any) => {
+              return <div key={heroes.id}>
+                <div className={styles.hero}>
+                  <Avatar alt="Photo" sx={{ width: 100, height: 100, margin: 5 }}
+                    className={styles.avatarStyle} src={heroes.thumbnail.path != null
+                      ? `${heroes.thumbnail.path}.${heroes.thumbnail.extension}` : 'Photo'} />
+                  <div className={styles.name}>{heroes.name}</div>
+                  <div className={styles.description} >{heroes.description}</div>
+                  <NavLink to={`/comics/${heroes.id}`} style={{ textDecoration: 'none' }}> {<Button variant="contained"
```
Comment (alinalaktionova) [inline src/HomePage.tsx:129]: choose 1 way of setting styles either inline or separate css files

Comment (alinalaktionova) [inline src/HomePage.tsx:129]: what's wrong with code style? linter should fix all this problems

File: src/NotFoundPage.tsx [hunk 16]
```diff
@@ -0,0 +1,16 @@
+import React from 'react';
+
+
```
Comment (alinalaktionova) [inline src/NotFoundPage.tsx:3]: what is this?

File: src/api.js [hunk 17]
```diff
@@ -0,0 +1,55 @@
...
+
+
+const PUBLIC_KEY = process.env.REACT_APP_API_KEY;
+const hash = process.env.REACT_APP_API_HASH;
+const baseURL = process.env.REACT_APP_BASEURL;
+
+const params = {
+  ts: '1',
+  apikey: PUBLIC_KEY,
+  hash: hash
```
Comment (alinalaktionova) [inline src/api.js]: no need to duplicate property with the same name

File: src/api.js [hunk 18]
```diff
@@ -0,0 +1,55 @@
...
+}
+
+export function getComicsByCharacter(characterId) {
+  return axios.get(`${baseURL}characters/${characterId}/comics?ts=1&apikey=${PUBLIC_KEY}&hash=${hash}`)
```
Comment (alinalaktionova) [inline src/api.js]: use params in axios for query

File: src/api.js [hunk 19]
```diff
@@ -0,0 +1,55 @@
...
+    .then(response => {
+ 
+      const res = response.data.data.results;
```
Comment (alinalaktionova) [inline src/api.js]: no need to create separate var for res

File: src/api.js [hunk 20]
```diff
@@ -0,0 +1,55 @@
...
+    })
+    .catch(error => {
+      console.log(error);
+    })
+
+}
+
+export function getCharacterByName(nameStartsWith) {
+
+  return axios.get(`${baseURL}characters?nameStartsWith=${nameStartsWith}&`, {params})
```
Comment (alinalaktionova) [inline src/api.js]: nameStartsWith can be in params object

File: src/assets/Spinner-1s-200px.svg [hunk 21]
```diff
@@ -0,0 +1,52 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (alinalaktionova) [inline src/assets/Spinner-1s-200px.svg]: you can just use library for spinner component))

**Review by alinalaktionova — CHANGES_REQUESTED** [review]

File: src/api.js (hunk 20 above)
Comment (krasyukden) [reply src/api.js]: can I leave it like this, because nameStartsWith comes in parameters?

File: src/assets/Spinner-1s-200px.svg (hunk 21 above)
Comment (krasyukden) [reply src/assets/Spinner-1s-200px.svg]: spinner was changed from MUI

File: src/ComicsPage.module.css [hunk 22]
```diff
@@ -1,5 +1,5 @@
 *{
-	margin: 0;
+    margin: 0;
 	padding: 0;
```
Comment (alinalaktionova) [inline src/ComicsPage.module.css]: padding also should have 2 spacing tab))

Comment (alinalaktionova) [reply src/ComicsPage.module.css]: and rename files please, there is a convention to name files from lowercase letter except react component files

**Review by alinalaktionova — CHANGES_REQUESTED** [review]
in general good, fix other remarks please

**Review by alinalaktionova — APPROVED** [review]

**Pull request merged**